### ISSUE TITLE
Make sure all pre-commit.ci errors are solved

### ISSUE THREAD

**Issue by shaypal5** [pulearn/pulearn#59]
The latest pre-commit.ci run on the latest commit in master failed with:

[WARNING] The 'rev' field of repo 'https://github.com/crate-ci/typos' appears to be a mutable reference (moving tag / branch).  Mutable references are never updated after first install and are not supported.  See https://pre-commit.com/#using-the-latest-version-for-a-repository for more details.  Hint: `pre-commit autoupdate` often fixes this.
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed
check json...............................................................Passed
check yaml...............................................................Passed
check toml...............................................................Passed
check docstring is first.................................................Passed
check that executables have shebangs.....................................Passed
check for case conflicts.................................................Passed
check for added large files..............................................Passed
detect private key.......................................................Passed
typos....................................................................Failed
- hook id: typos
- exit code: 2

error: `inpurt` should be `input`
  --> src/pulearn/elkanoto.py:120:44
    |
120 |             Predicted labels for the given inpurt samples.
    |                                            ^^^^^^
    |
error: `inpurt` should be `input`
  --> src/pulearn/elkanoto.py:283:44
    |
283 |             Predicted labels for the given inpurt samples.
    |                                            ^^^^^^
    |
error: `permut` should be `permute`
  --> examples/BreastCancerElkanotoExample.py:45:5
   |
45 |     permut = np.random.permutation(len(y))
   |     ^^^^^^
   |
error: `permut` should be `permute`
  --> examples/BreastCancerElkanotoExample.py:46:11
   |
46 |     X = X[permut]
   |           ^^^^^^
   |
error: `permut` should be `permute`
  --> examples/BreastCancerElkanotoExample.py:47:11
   |
47 |     y = y[permut]
   |           ^^^^^^
   |

docformatter.............................................................Passed
mdformat.................................................................Passed
prettier.................................................................Passed
Black by Ruff............................................................Passed
Ruff check...............................................................Failed
- hook id: ruff
- exit code: 1

warning: `incorrect-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible. Ignoring `incorrect-blank-line-before-class`.
warning: `multi-line-summary-first-line` (D212) and `multi-line-summary-second-line` (D213) are incompatible. Ignoring `multi-line-summary-second-line`.
notebooks/feature_importance_for_bagging.ipynb:cell 22:6:80: E501 Line too long (98 > 79)
  |
5 | # # Your code to create the ensemble bagging model
6 | # xgb = XGBClassifier(use_label_encoder=False, objective='binary:logistic', eval_metric='logloss')
  |                                                                                ^^^^^^^^^^^^^^^^^^^ E501
7 | # pu_estimator = BaggingClassifier(estimator=xgb, n_estimators=15)
8 | # pu_estimator.fit(X, Y)
  |

notebooks/feature_importance_for_bagging.ipynb:cell 22:17:5: F821 Undefined name `pu_estimator`
   |
16 | average_feature_importances = feature_importances_sum / len(
17 |     pu_estimator.estimators_
   |     ^^^^^^^^^^^^ F821
18 | )
   |

notebooks/feature_importance_for_bagging.ipynb:cell 22:21:1: E402 Module level import not at top of cell
   |
20 | # Visualize average feature importances
21 | import matplotlib.pyplot as plt
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ E402
22 |
23 | plt.bar(range(len(average_feature_importances)), average_feature_importances)
   |

src/pulearn/bagging.py:37:23: A004 Import `zip` is shadowing a Python builtin
   |
35 | from joblib import Parallel, delayed
36 | from six import with_metaclass
37 | from six.moves import zip
   |                       ^^^ A004
38 | from sklearn.base import ClassifierMixin
39 | from sklearn.metrics import accuracy_score
   |

Found 4 errors.

pyproject-fmt............................................................Passed
Validate pyproject.toml..................................................Passed
